src/javascript/waxeye.js

```diff
@@ -379,22 +353,33 @@
                   return MachineState.INTER(MachineConfiguration.EVAL(exp.args[0], pos, [], err, arrayPrepend(Continuations.CONT_VOID(asts), k)));
                 case "CHAR":
                   c = exp.args[0];
-                  return MachineState.INTER(MachineConfiguration.APPLY(k, c.length === 1 ? (eof(pos)) || (c !== input[pos]) ? Value.FAIL(updateError(err, pos, new ErrChar(c))) : Value.VAL(pos + 1, arrayPrepend(input[pos], asts), err) : eof(pos + 1) || c[0] !== input[pos] || c[1] !== input[pos + 1] ? Value.FAIL(updateError(err, pos, new ErrChar(c))) : Value.VAL(pos + 2, arrayPrepend(input[pos] + input[pos + 1], asts), err)));
+                  return MachineState.INTER(MachineConfiguration.APPLY(k, c.length === 1 ? (eof(pos)) || (c !== input[pos]) ? Value.FAIL(updateError(err, pos, new ErrChar(c))) : Value.VAL(pos + 1, arrayPrepend(input[pos], asts), err) : eof(pos + 1) || c[0] !== input[pos] || c[1] !== input[pos + 1] ? Value.FAIL(updateError(err, pos, new ErrChar(c))) : Value.VAL(pos + 2, arrayPrepend(input[pos] + input[pos + 1], asts), err))); // c.length == 2
                 case "CHAR_CLASS":
                   cc = exp.args;
                   if (eof(pos)) {
                     return MachineState.INTER(MachineConfiguration.APPLY(k, Value.FAIL(updateError(err, pos, new ErrCC(cc)))));
                   }
+                  // JavaScript string comparison does not compare Unicode characters correctly, so we must compare
+                  // codepoints. Example:
+                  //   "ﬆ" > '𝌆' //=> true
+                  //   "ﬆ".codePointAt(0) > '𝌆'.codePointAt(0) //=> false
                   inputCodePoint = input.codePointAt(pos);
+                  // Loop over cc instead of recursing to avoid stack overflow on large character classes.
                   for (j = 0, len = cc.length; j < len; j++) {
                     charClass = cc[j];
+                    // Single character
```

Comment:
Is something missing here? If not, the comments are misleading

Reply:
This is a generated file.